docs/tasks-uses-cases.mdx

```diff
@@ -0,0 +1,161 @@
+    activate LLM
+
+    %% Second dependency task
+    LLM->>Client: Generate code (needs dependency: Lodash)
+    deactivate LLM
+    Client->>Server: tools/call "ImportDependency"<br/>name: "Lodash"<br/>task: { ttl: 300000 }
+    Server->>Client: CreateTaskResult<br/>{ taskId: "dep-002",<br/>status: "working" }
+    Client->>LLM: Task created for Lodash<br/>(taskId: dep-002)
+    activate LLM
+
+    Note over LLM: LLM continues generating code<br/>(avoiding dependency-specific code)
+
+    LLM->>Client: Generate utility functions<br/>(no dependencies needed)
+    deactivate LLM
+    Client->>LLM: Files updated
+    activate LLM
+
+    Note over Server,Approval: Background: Approval workflows begin
+
+    Server->>Approval: Request approval: Axios
+    activate Approval
+    Server->>Approval: Request approval: Lodash
+    activate Approval
+
+    Note over Client: Client begins polling task statuses
+
+    %% Client polls while work is in progress
+    Client->>Server: tasks/get<br/>taskId: "dep-001"
+    Server->>Client: status: "working"<br/>(approval pending)
+
+    Client->>Server: tasks/get<br/>taskId: "dep-002"
+    Server->>Client: status: "working"<br/>(approval pending)
+
+    Note over LLM: LLM continues generating more code
+
+    Note over Approval,Server: Lodash approval completes
+    Approval-->>Server: Approved ✓
+    deactivate Approval
+    Note right of Server: Task dep-002<br/>status → "completed"
+
+    %% Client polls again and finds Lodash completion
+    Client->>Server: tasks/get<br/>taskId: "dep-001"
+    Server->>Client: status: "working"<br/>(approval still pending)
+
+    Client->>Server: tasks/get<br/>taskId: "dep-002"
+    Server->>Client: status: "completed"
+
+    Client->>Server: tasks/result<br/>taskId: "dep-002"
+    Server->>Client: Result:<br/>{ approved: true,<br/>dependency: "Lodash" }
+
+    Client->>LLM: [System Message]<br/>Dependency approved: Lodash<br/>You may now use it
+
+    Note over LLM: LLM can now generate code<br/>using Lodash
+
+    LLM->>Client: Generate code using Lodash
+    deactivate LLM
+    Client->>LLM: Files updated
+    activate LLM
+
+    Note over Approval,Server: Axios approval completes
+    Approval-->>Server: Rejected ✗
+    deactivate Approval
+    Note right of Server: Task dep-001<br/>status → "completed"
+
+    %% Check second task
+    Client->>Server: tasks/get<br/>taskId: "dep-001"
+    Server->>Client: status: "completed"
+
+    Client->>Server: tasks/result<br/>taskId: "dep-001"
+    Server->>Client: Result:<br/>{ approved: false,<br/>dependency: "Axios",<br/>reason: "Security policy violation" }
+
+    Client->>LLM: [System Message]<br/>Dependency rejected: Axios<br/>Reason: Security policy violation<br/>Find alternative solution
+
+    Note over LLM: LLM must find alternative<br/>to Axios (or give up)
+
+    LLM->>Client: Generate code using fetch API<br/>(built-in alternative to Axios)
+    deactivate LLM
+    Client->>LLM: Files updated
+    activate LLM
+```
```

Comment:
`Client begins polling task statuses` need some interval, the current shape seems always polling in a while loop